libraries/esp8266/examples/LowPowerDemo/LowPowerDemo.ino

```diff
@@ -0,0 +1,338 @@
+   preferably not the 3V3 pin on the module or it adds to the measured amperage.  When the
+   LED blinks you can proceed to the next test.  When the LED is lit continuously it's
+   connecting WiFi, and when it's off the CPU is asleep.  The LED blinks slowly when the
+   tests are complete.  Test progress can also be shown on the serial monitor.
+
+   WiFi connections will be made over twice as fast if you can use a static IP address.
+
+   This example is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   This example is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this example; if not, write to the Free Software Foundation, Inc.,
+   51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA  */
+
+#include <ESP8266WiFi.h>
+#include <coredecls.h>         // crc32()
+#include <PolledTimeout.h>
+
+//#define DEBUG  // prints WiFi connection info to serial, uncomment if you want WiFi messages
+#ifdef DEBUG
+#define DEBUG_PRINTLN(x)  Serial.println(x)
+#define DEBUG_PRINT(x)  Serial.print(x)
+#else
+#define DEBUG_PRINTLN(x)
+#define DEBUG_PRINT(x)
+#endif
+
+#define WAKE_UP_PIN 0  // D3/GPIO0, can also force a serial flash upload with RESET
+// you can use any pin for WAKE_UP_PIN except for D0/GPIO16 as it doesn't support interrupts
+
+// uncomment one of the two lines below for your LED connection, if used
+#define LED 5  // D1/GPIO5 external LED for modules with built-in LEDs so it doesn't add to the amperage
+//#define LED 2  // D4/GPIO2 LED for ESP-01,07 modules; D4 is LED_BUILTIN on most other modules
+// you can use LED_BUILTIN, but it adds to the measured amperage by 0.3mA to 6mA.
+
+ADC_MODE(ADC_VCC);  // allows you to monitor the internal VCC level; it varies with WiFi load
+// don't connect anything to the analog input pin(s)!
+
+// enter your WiFi configuration below
+const char* AP_SSID = "SSID";  // your router's SSID here
+const char* AP_PASS = "password";  // your router's password here
+IPAddress staticIP(0, 0, 0, 0); // parameters below are for your static IP address, if used
+IPAddress gateway(0, 0, 0, 0);
+IPAddress subnet(0, 0, 0, 0);
+IPAddress dns1(0, 0, 0, 0);
+IPAddress dns2(0, 0, 0, 0);
+uint32_t wifiTimeout = 25E3;  // 25 second timeout on the WiFi connection
+
+// This structure will be stored in RTC memory to remember the reset count (number of Deep Sleeps).
+// First field is CRC32, which is calculated based on the rest of the structure contents.
+// Any fields can go after the CRC32.  The structure must be 4-byte aligned.
+struct {
```

Comment:
If I'm reading your code correctly, I think the definition in the struct needs to change since resetCount is now an int, so
```cpp
byte _data[4];  // the last byte stores the reset count
```
should be something like
```cpp
int _data[];  // 4 bytes for the int resetCount
```
with matching revisions to readFrom and writeTo